apps/api/src/app/integrations/usecases/update-integration/update-integration.usecase.ts

```diff
@@ -0,0 +1,49 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { IntegrationEntity, IntegrationRepository } from '@notifire/dal';
+import { UpdateIntegrationCommand } from './update-integration.command';
+
+@Injectable()
+export class UpdateIntegration {
+  constructor(private integrationRepository: IntegrationRepository) {}
+
+  async execute(command: UpdateIntegrationCommand): Promise<IntegrationEntity> {
+    const existingIntegration = await this.integrationRepository.findById(command.integrationId);
+    if (!existingIntegration) throw new NotFoundException(`Entity with id ${command.integrationId} not found`);
+
+    const updatePayload: Partial<IntegrationEntity> = {};
+    if (command.providerId) {
+      updatePayload.providerId = command.providerId;
+    }
+
+    if (command.channel) {
+      updatePayload.channel = command.channel;
+    }
+
+    if (command.active) {
+      updatePayload.active = command.active;
+    }
+
+    if (command.credentials) {
+      updatePayload.credentials = command.credentials;
+    }
+
+    if (!Object.keys(updatePayload).length) {
+      throw new BadRequestException('No properties found for update');
+    }
+
+    await this.integrationRepository.update(
+      {
+        _id: command.integrationId,
+        active: command.active,
+        _applicationId: command.applicationId,
+        channel: command.channel,
+        providerId: command.providerId,
+      },
```

Comment:
```suggestion
      {
        _id: command.integrationId,
        _applicationId: command.applicationId
      },
```
There is no need to specify so much the query for update, the _applicationId is important to make sure that the authenticated user has the credentials to access this specific entity.

Reply:
Pushed a PR regarding minimalizing the amount of params on update query https://github.com/notifirehq/notifire/pull/342